admin/database/database.go

```diff
@@ -573,19 +594,21 @@ func (d DeploymentStatus) String() string {
 // Deployment is a single deployment of a git branch.
 // Deployments belong to a project.
 type Deployment struct {
-	ID                string           `db:"id"`
-	ProjectID         string           `db:"project_id"`
-	OwnerUserID       *string          `db:"owner_user_id"`
-	Environment       string           `db:"environment"`
-	Branch            string           `db:"branch"`
-	RuntimeHost       string           `db:"runtime_host"`
-	RuntimeInstanceID string           `db:"runtime_instance_id"`
-	RuntimeAudience   string           `db:"runtime_audience"`
-	Status            DeploymentStatus `db:"status"`
-	StatusMessage     string           `db:"status_message"`
-	CreatedOn         time.Time        `db:"created_on"`
-	UpdatedOn         time.Time        `db:"updated_on"`
-	UsedOn            time.Time        `db:"used_on"`
+	ID                     string                  `db:"id"`
+	ProjectID              string                  `db:"project_id"`
+	OwnerUserID            *string                 `db:"owner_user_id"`
+	Environment            string                  `db:"environment"`
+	Branch                 string                  `db:"branch"`
+	RuntimeHost            string                  `db:"runtime_host"`
+	RuntimeInstanceID      string                  `db:"runtime_instance_id"`
+	RuntimeAudience        string                  `db:"runtime_audience"`
+	Status                 DeploymentStatus        `db:"status"`
+	DesiredStatus          DeploymentDesiredStatus `db:"desired_status"`
+	StatusMessage          string                  `db:"status_message"`
+	CreatedOn              time.Time               `db:"created_on"`
+	UpdatedOn              time.Time               `db:"updated_on"`
+	UsedOn                 time.Time               `db:"used_on"`
+	DesiredStatusUpdatedOn time.Time               `db:"desired_status_updated_on"`
```

Comment:
Instead of tracking this, can the reconciling job instead have a check like this at the end:
```
if depl.Status != depl.DesiredStatus {
  // The desired state changed while the job executed.
  // Re-run the job.
  ...
}
```

Reply:
Would be clean, but how would we detect two consecutive calls to update project, where one happened during the job execution? The desired state wouldn't change in that scenario.